src/shared/killall.c

```diff
@@ -38,6 +40,20 @@ static bool ignore_proc(pid_t pid, bool warn_rootfs) {
         if (r != 0)
                 return true; /* also ignore processes where we can't determine this */
 
+        /* Ignore processes that are part of a cgroup marked with the trusted.survive xattr */
+        r = cg_pid_get_path_shifted(pid, /* root= */ NULL, &cgroup_path);
+        if (r < 0)
+                log_warning_errno(r, "Failed to get cgroup path of process " PID_FMT ", ignoring: %m", pid);
+        else {
+                r = cg_get_xattr_bool(SYSTEMD_CGROUP_CONTROLLER, cgroup_path, "trusted.survive");
+                if (r < 0 && !ERRNO_IS_XATTR_ABSENT(r))
+                        log_debug_errno(r,
+                                        "Failed to get trusted.survive xattr of cgroup %s, ignoring: %m",
+                                        cgroup_path);
+                if (r > 0)
+                        return true;
+        }
+
```

Comment:
ignore_proc() is growing too large for my taste. Any chance you can split this block out?

(and maybe the argv[0][0] check below too